ConwaysGameOfLife/Generator.cs

```diff
@@ -28,13 +32,39 @@ public Generator()
 
         */
 
-        public bool ShouldSwitchCellState()
+        public Universe GenerateNewUniverse()
+        {
+            var oldCells = _oldUniverse.Cells;
+            var newCells = new List<Cell>();
+            foreach (var cell in oldCells)
+            {
+                GenerateCells(newCells, cell);
+            }
+            return new Universe(_oldUniverse.GridWidth, _oldUniverse.GridLength, newCells);
+        }
+
+        private void GenerateCells(List<Cell> cells, Cell cell)

```

Comment:
this function can return a list of cells as it's what the function name is suggesting.